sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsOptions.java

```diff
@@ -33,11 +34,23 @@
 
   /** AWS region used by the AWS client. */
   @Description("AWS region used by the AWS client")
-  @Validation.Required
+  @Default.InstanceFactory(AwsRegionFactory.class)
   String getAwsRegion();
 
   void setAwsRegion(String value);
 
+  /** Attempt to load default region. */
+  class AwsRegionFactory implements DefaultValueFactory<String> {
+    @Override
+    public String create(PipelineOptions options) {
+      try {
+        return new DefaultAwsRegionProviderChain().getRegion();
+      } catch (SdkClientException e) {
+        return null;
```

Comment:
Should we print out an error here? Can it cause NPE later?

Reply:
No, this should be silent ... if no region is found, the option should be `null` as it is at the moment in this case. Same regarding the NPE, this just sets a region automatically if available. Otherwise the behavior is unchanged.
Btw, check this comment in the AWS SDK itself:
```
// Note: This is a bug in the provider chain. The chain should return null when no region is found according to
 // the interface, but an exception is thrown  here instead. This class is used in too many places to change now.
// TODO: In 2.0, be sure this bug does not carry through.
 throw new SdkClientException("Unable to load region information from any provider in the chain"
```

Reply:
Ok, thanks